File: recursion-tutorial-fc/Palindrome.java
public class Palindrome {

    public static boolean isPalindrome(String str) {

        if(str.length() == 1 || str.length() == 0) return true;

        if(str.charAt(0) == str.charAt(str.length() - 1)) {

            return isPalindrome(str.substring(1, str.length() - 1));
        }

        return false;

    }

    public static boolean isPalindrome2(String str) {

        int i = 0;
        int j = str.length() - 1;

        while(i < j) {

            if(str.charAt(i) == str.charAt(j)) {

                i++;
                j--;
            }else {

                return false;
            }
        }

        return true;
    }

    public static void main(String[] args) {

        System.out.println(Palindrome.isPalindrome("kajak"));
        System.out.println(Palindrome.isPalindrome2("kajak"));
    }
}

File: recursion-tutorial-fc/palindrome.js
const isPalindrome = (string) => {
    
    if(string.length == 0 || string.length == 1) return true;

    if(string.charAt(0) == string.charAt(string.length - 1)) {

        return isPalindrome(string.substring(1, string.length - 1));
    }

    return false;
}

const isPalindrome2 = (string) => {
    
    if(string.length == 0 || !string) return false;
    if(string.length == 1) return true;

    let i = 0, j = string.length -1;
    while( i < j) {

        if(string.charAt(i) == string.charAt(j)) {
            i++;
            j--;
        }else {
            return false;
        }
    }

    return true;
}

// isPalindrome('abcd');
console.log(isPalindrome('abcd'));
console.log(isPalindrome2('kaak'));